### ISSUE TITLE
[Bug]: Upload icon vanishes at times even when there are uploads pending to complete

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#5847]
### Summary

The upload queue icon that we now have with latest app revamp seems to vanish at times even though there are uploading pending to be completed. It makes it difficult to manage the ongoing uploads.

If we couldn't figure out what's causing this, I suppose we could at least do a stopgap solution and just show the upload queue icon all the time to avoid confusion.

### Steps to reproduce

1. Initiate some uploads in the app (I initiated the upload by sharing the image to the app)
2. Complete the details necessary for the upload.
3. Go to the Contributions screen of the app.

### Expected behaviour

The upload queue should always be visible making it easy to manage ongoing uploads.

### Actual behaviour

The upload queue icon flickers and is sometimes not visible making it difficult to access the queue and manage ongoing uploads.

### Device name

OnePlus Nord

### Android version

Android 12

### Commons app version

main

### Screen-shots

- [Screen record showing the icon flcikering](https://drive.google.com/file/d/15XPLcJ-6nZIF6ISHY3_rJ8E9D84XwO3L/view?usp=drive_link)
- [Screen record showing the icon only appearing after an app reopen](https://drive.google.com/file/d/12lD7vPH2Uj2NYMwgbhNo59Oa2MvgKeFJ/view?usp=drive_link)

### Would you like to work on the issue?

None

**Comment by nicolas-raoul** [comment]
I just tried an upload of 8 pictures, then 10. Strangely for me the icon is always correctly present during the whole upload. Even when restarting the app in the middle...

**Comment by bxy379987** [comment]
@nicolas-raoul sir, can you assign this issue to me, I and my friend are ANU students, and our teacher tell us that we need to find issue to solve in this repo, we think this issue is not too hard and will try our best to solve this

**Comment by nicolas-raoul** [comment]
@bxy379987 Are you able to reproduce the issue? If yes, let us know and you can have it. Thanks! :-)

**Comment by bxy379987** [comment]
I face some problems when gradle build, I will try to reproduce, if I success I will tell you

**Comment by bxy379987** [comment]
@nicolas-raoul 
![1728962629851](https://github.com/user-attachments/assets/e3831963-07f3-4554-a1b9-c5957781d62e)
Sir I have uploaded a picture to this common app, when i uploaded it, I found that as the issue said, the Contributions screen of this app had no upload icon, i don't know if my upload progress is working, but after about 10 sec, it shows that my picture is successfully uploaded.

**Comment by nicolas-raoul** [comment]
Did you see the upload icon at least a little bit? At the left of the the bell icon.
Would you mind posting a screencast showing the issue reproduction?
Thanks a lot!

**Comment by bxy379987** [comment]
Yes i see,
![1728983862026](https://github.com/user-attachments/assets/d9ec2050-dd7b-4ef6-8fe6-e5846c0c7621)
 and I found that when the upload icon on the contribution interface does not display, you can simply switch to another interface, like nearby, and then switch back to contribution to make the icon reappear, without needing to restart the app.

**Comment by bxy379987** [comment]
But I don't have a clue how to fix this bug now.,I think the problem should be in the upload package.,But his code seems to be automatically injected.,So there's no inexplicable situation where the component becomes invisible.,Can you give me some suggestions.,How should I modify it and so on.

**Comment by nicolas-raoul** [comment]
I would suggest inserting logging and breakpoints at the methods that hide and show the icon.
Then see when it gets unduly hidden, and find out why.
Thanks! :-)

**Comment by bxy379987** [comment]
got it，thank you！

**Comment by nicolas-raoul** [comment]
I believe you can take several pictures with the emulator's camera, then select them all in Android's stock gallery app, then share them to the Commons app.

**Comment by bxy379987** [comment]
[apps-android-commons – ContributionsFragment.java [apps-android-commons.app.main] 2024-10-16 15-10-18.zip](https://github.com/user-attachments/files/17389077/apps-android-commons.ContributionsFragment.java.apps-android-commons.app.main.2024-10-16.15-10-18.zip)
It seems when I upload mutiple pictures, the upload icon works well, I only change a little bit of code and most of them are adding logs, it is strange, I will try more times later with my group members

**Comment by bxy379987** [comment]
[apps-android-commons – ContributionsFragment.java [apps-android-commons.app.main] 2024-10-16 15-18-45.zip](https://github.com/user-attachments/files/17389195/apps-android-commons.ContributionsFragment.java.apps-android-commons.app.main.2024-10-16.15-18-45.zip)
I think it might be because I didn't upload under specific conditions, so the icon flickering issue wasn't triggered. I would like to ask under what conditions you encountered this icon flickering issue. I also tried uploading multiple images, and it wasn't the first upload after reopening the app. Instead, I uploaded several times after opening the app, and this solved the issue.

**Comment by nicolas-raoul** [comment]
@bxy379987 It seems like it does not happen for everyone. If anyone in your group is able to reproduce, they can work on this. If nobody in your group is able to reproduce, would you mind letting us know? Then the issue can be handled by someone who can reproduce.

Anyone has reproduced the issue? If yes please say your Android version, whether emulator or not, etc.

@sivaraam This happened on a physical device running Android 12, more environment details perhaps, maybe commit number shown in About?

**Comment by bxy379987** [comment]
I have shared this issue with my teammates and if they can I will tell you,  meantime I will look for other issues that fit for me, since i am familiar with this project a little bit :).

**Comment by junnzhuu** [comment]
Unable to reproduce this issue. I am using a Redmi K30 Pro with Android 12, and the icon displays normally.

**Comment by bxy379987** [comment]
Yes, it seems that we are not having this issue. Another team member hasn't responded to me yet, but it seems like he has some ideas for modifying this part of the code. I'll go ask him again.

**Comment by nicolas-raoul** [comment]
@sivaraam Do you still see this in the latest main? I remember seeing this issue a while ago, but never recently. :-)

**Comment by sivaraam** [comment]
> @sivaraam This happened on a physical device running Android 12, more environment details perhaps, maybe commit number shown in About?

Yes. This happened on a physical device running Android 12. The commit ID I see in the "About" is 5.1.0~f47a0dd49 which is f47a0dd49

> @sivaraam Do you still see this in the latest main? I remember seeing this issue a while ago, but never recently. :-)

It still vanishes for me when I have the queue paused and fails to show up when opening media from the "Contributions" screen. I suppose "Pause-Resume" is a certain factor to trigger this. It starts flickering when there are multiple uploads and it is close to uploading the last one.

Screen recordings:

- [Icon disappear on pause](https://drive.google.com/file/d/19ZANKxbfDIbdMTee8dZe_IGfQ_k4GmW2/view?usp=sharing)
- [Icon flicker when nearing last upload of multi upload](https://drive.google.com/file/d/12KStlQXDcIeBO3Y44H5VHU7qxg-94O9J/view?usp=sharing)

**Comment by sivaraam** [comment]
If reproduction seems too difficult, why not just make the icon always visible and just adjust the count accordingly as and when new uploads are queued / completed? I think that would help avoid the situation where the upload management screen is inaccessible at some particular time when it should be.

PS: Just for clarity, I'm not really waiting for the fix for this to release the beta. I'm still looking to publish the beta out as soon as I can.

**Comment by nicolas-raoul** [comment]
Indeed, sounds reasonable to show it always.
@bxy379987 Do you want to try?

**Comment by bxy379987** [comment]
According to your suggestion, "If reproduction seems too difficult, why not just make the icon always visible and just adjust the count accordingly as and when new uploads are queued / completed," I have set the upload icon to always be visible, and I have also included some logs from when I was debugging.

This is not the final version of the PR; I just wanted to show you my current modifications. To be honest, I feel that my changes, which only involve setting the upload icon to always visible in one method, might meet your requirement but seem superficial and possibly not in line with coding standards. I am not very satisfied with this modification. However, due to limitations in my skills and time, I haven't found a better method yet.

If you think it useful, I can delete all the logs and submit again. I apologize for not being able to successfully reproduce and resolve this issue. Perhaps you might consider finding someone else. I'm sorry for wasting your time.
https://github.com/commons-app/apps-android-commons/pull/5896

**Comment by sivaraam** [comment]
> If you think it useful, I can delete all the logs and submit again. I apologize for not being able to successfully reproduce and resolve this issue. Perhaps you might consider finding someone else. I'm sorry for wasting your time.

You do not have to take it too personally. The issue was apparently not reproducible in all devices. So, it is not a problem that you were not able to reproduce the same. 🙂 

Thank you very much for working on this issue and for quickly providing a fix for making the icon always visible.

**Comment by nicolas-raoul** [comment]
I just got this on 5.0.2-debug-rohit9625-revert-commit~844437389

https://github.com/user-attachments/assets/b1da00ac-f448-414e-aa17-5a9a422a9c20

**Comment by nicolas-raoul** [comment]
Anyone willing to continue work on [this pull request](https://github.com/commons-app/apps-android-commons/pull/5901)? :-)
Just create a branch and pull that pull request's branch into yours, then send as a different pull request.
Thanks!

**Comment by parneet-guraya** [comment]
Let me do this :)
